### PR TITLE
v7.0.0-beta.7

### PR DESCRIPTION
**Pull request by alexfauquette**
### Release the packages

1. Checkout the last version of the working branch
2. `yarn && yarn release:build` (make sure you have the latest dependencies installed, and build the packages)
3. `yarn release:publish` (release the versions on npm, you need your 2FA device)
4. `yarn release:tag` (push the newly created tag)

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```bash
git push -f upstream next:docs-next
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-next)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-next` deployment.

### CODE REVIEW

**Review by noraleonte — APPROVED** [review]
LGTM 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,102 @@
...
+- The `columnHeader--showColumnBorder` class was replaced by `columnHeader--withLeftBorder` and `columnHeader--withRightBorder`.
+- The `columnHeadersInner`, `columnHeadersInner--scrollable`, and `columnHeaderDropZone` classes were removed since the inner wrapper was removed in our effort to simplify the DOM structure and improve accessibility.
+- The `pinnedColumnHeaders`, `pinnedColumnHeaders--left`, and `pinnedColumnHeaders--right` classes were removed along with the element they were applied to.
+  The pinned column headers now use `position: 'sticky'` and are rendered in the same row element as the regular column headers.
+
+#### `@mui/x-data-grid@7.0.0-beta.7`
+
+- [DataGrid] Make `rowCount` part of the state (#12381) @MBilalShafi
+- [DataGrid] Make column resizing and autosizing available in Community plan (#12420) @cherniavskii
+- [DataGrid] Remove baseSwitch (#12439) @romgrk
```
Comment (LukasTy) [inline CHANGELOG.md:30]:
```suggestion
- [DataGrid] Remove `baseSwitch` slot (#12439) @romgrk
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,102 @@
...
+- [docs] Polish What's new in MUI X blog titles (#12309) @oliviertassinari
+- [docs] Replace `rel="noreferrer"` by `rel="noopener"` @oliviertassinari
+- [docs] Update `date-fns` `weekStarsOn` overriding example (#12416) @LukasTy
+
+### Core
+
+- [core] Fix CI (#12414) @flaviendelangle
+- [core] Fix PR deploy link for Tree View doc pages (#12411) @flaviendelangle
+- [data grid] Fix focus visible style on scrollbar (#12402) @oliviertassinari
+- [data grid] Fixed issue where pressing Delete key resets various cell values to empty string. (#12216) @sooster910
```
Comment (LukasTy) [inline CHANGELOG.md:100]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,102 @@
...
+
+#### Breaking changes
+
+- The `columnHeader--showColumnBorder` class was replaced by `columnHeader--withLeftBorder` and `columnHeader--withRightBorder`.
+- The `columnHeadersInner`, `columnHeadersInner--scrollable`, and `columnHeaderDropZone` classes were removed since the inner wrapper was removed in our effort to simplify the DOM structure and improve accessibility.
+- The `pinnedColumnHeaders`, `pinnedColumnHeaders--left`, and `pinnedColumnHeaders--right` classes were removed along with the element they were applied to.
+  The pinned column headers now use `position: 'sticky'` and are rendered in the same row element as the regular column headers.
+
+#### `@mui/x-data-grid@7.0.0-beta.7`
+
```
Comment (LukasTy) [inline CHANGELOG.md:27]:
```suggestion

- [DataGrid] Fix focus visible style on scrollbar (#12402) @oliviertassinari
- [DataGrid] Fix the issue where pressing the Delete key resets various cell values to an empty string. (#12216) @sooster910
```

**Review by flaviendelangle — APPROVED** [review]

**Review by LukasTy — APPROVED** [review]

**Pull request merged**